provider/cluster/kube/k8s_integration_test.go

```diff
@@ -89,6 +92,39 @@ func TestNewClient(t *testing.T) {
 	assert.NoError(t, err)
 	assert.NotEqual(t, maxtries, tries)
 
+	logs, err := client.ServiceLogs(ctx, lid, svcname, true, nil)
+	require.NoError(t, err)
+	require.Equal(t, int(sstat.AvailableReplicas), len(logs))
+
+	log := make(chan string)
+	defer close(log)
+
+	var wgFinished sync.WaitGroup
+	wgFinished.Add(1)
+	go func(scan *bufio.Scanner) {
+		defer wgFinished.Done()
+
+		for scan.Scan() {
+			lg := scan.Text()
+			fmt.Println(lg)
+			log <- lg
+			break
+		}
+	}(logs[0].Scanner)
+
+	select {
+	case line := <-log:
+		assert.NotEmpty(t, line)
+	case <-time.After(10 * time.Second):
+		_ = logs[0].Stream.Close()
+		assert.Fail(t, "timed out waiting for logs")
+	}
+
+	for _, lg := range logs {
+		assert.NoError(t, lg.Stream.Close(), lg.Name)
+	}
+
+	// logs[0].Stream.
```

Comment:
The [same](https://github.com/ovrclk/akash/pull/721#discussion_r448777841) race condition is still here - `log <- lg` can get stuck.